Use 'CanvasText' instead of 'text' in the UA stylesheet

pxlcoder opened the pull request:
#### 58a8fc782a7d3dd00aedf6bd5054c51f1dd5d412
<pre>
Use &apos;CanvasText&apos; instead of &apos;text&apos; in the UA stylesheet
<a href="https://bugs.webkit.org/show_bug.cgi?id=232954">https://bugs.webkit.org/show_bug.cgi?id=232954</a>
rdar://85510727

Reviewed by Wenson Hsieh.

&apos;CanvasText&apos; is a standard system color which should be used for &quot;text in
application content or documents&quot;.

Spec: <a href="https://www.w3.org/TR/css-color-4/#valdef-system-color-canvastext">https://www.w3.org/TR/css-color-4/#valdef-system-color-canvastext</a>

* Source/WebCore/css/html.css:

Use the matching &apos;Canvas&apos; value alongside &apos;CanvasText&apos;, rather than the
non-standard &apos;-webkit-control-background&apos;.

This is a slight behavior change on macOS, as &apos;CanvasText&apos; is implemented using
&apos;-[NSColor textBackgroundColor]&apos;, while &apos;-webkit-control-background&apos; is
implemented using &apos;-[NSColor controlBackgroundColor]&apos;. However, AppKit actually
uses &apos;-[NSColor textBackgroundColor]&apos; for text-based controls.
&apos;-[NSColor controlBackgroundColor]` is intended for &quot;content areas&quot;, such as
scroll views, table views, and collection views.

On other platforms, &apos;CanvasText&apos; and &apos;-webkit-control-background&apos; are equivalent.

Canonical link: <a href="https://commits.webkit.org/252845@main">https://commits.webkit.org/252845@main</a>
</pre>

nt1m reviewed (commented):
I think stylesheets should use background/text color pairs.

e.g. Canvas/CanvasText. Button/ButtonText. etc. They're guaranteed to work regardless of what's done on system colors

pxlcoder committed:
Use 'CanvasText' instead of 'text' in the UA stylesheet
https://bugs.webkit.org/show_bug.cgi?id=232954
rdar://85510727

Reviewed by Wenson Hsieh.

'CanvasText' is a standard system color which should be used for "text in
application content or documents".

Spec: https://www.w3.org/TR/css-color-4/#valdef-system-color-canvastext

* Source/WebCore/css/html.css:

Use the matching 'Canvas' value alongside 'CanvasText', rather than the
non-standard '-webkit-control-background'.

This is a slight behavior change on macOS, as 'CanvasText' is implemented using
'-[NSColor textBackgroundColor]', while '-webkit-control-background' is
implemented using '-[NSColor controlBackgroundColor]'. However, AppKit actually
uses '-[NSColor textBackgroundColor]' for text-based controls.
'-[NSColor controlBackgroundColor]` is intended for "content areas", such as
scroll views, table views, and collection views.

On other platforms, 'CanvasText' and '-webkit-control-background' are equivalent.

Canonical link: https://commits.webkit.org/252845@main
